api/src/services/generate-xform.js

```diff
@@ -16,17 +16,27 @@ const MEDIA_SRC_ATTR = ' data-media-src="';
 
 const FORM_STYLESHEET = path.join(__dirname, '../xsl/openrosa2html5form.xsl');
 const MODEL_STYLESHEET = path.join(__dirname, '../../node_modules/enketo-xslt/xsl/openrosa2xmlmodel.xsl');
+const XSLTPROC_CMD = 'xsltproc';
 
 const transform = (formXml, stylesheet) => {
   return new Promise((resolve, reject) => {
-    const xsltproc = childProcess.spawn('xsltproc', [ stylesheet, '-' ]);
+    const xsltproc = childProcess.spawn(XSLTPROC_CMD, [ stylesheet, '-' ]);
     let stdout = '';
     let stderr = '';
     xsltproc.stdout.on('data', data => stdout += data);
     xsltproc.stderr.on('data', data => stderr += data);
     xsltproc.stdin.setEncoding('utf-8');
-    xsltproc.stdin.write(formXml);
-    xsltproc.stdin.end();
+    try {
+      xsltproc.stdin.write(formXml);
+    } catch (err) {
+      if (err.code === 'EPIPE') {
+        return reject(new Error(`Command ${XSLTPROC_CMD} not found`));
+      }
+      logger.error(err);
```

Comment:
Yes there is no unique message to identify the lack of the command, nor simple way in Node to check first whether the command exist or not in the `$PATH` (that vary regarding the OS).

The `EPIPE` error quote from the Node docs is: _A write on a pipe, socket, or FIFO for which there is no process to read the data_, and because the command does not exist there is no process to read as the definition says.

However I found another way to raise the same exception, that is changing the execution permission of the command, e.g. to make it only executable by root or a different user that the API process. It's unlikely this error to happen but I like the message you propose, it's is a more broad error message that makes anyway clear that the problem is the availability of the command. Changing ...

Reply:
> that is changing the execution permission of the command...

Oh that's very good thing to know! Thanks for sharing.